docs/documentation/best-practices.md

```diff
@@ -14,7 +14,8 @@ Only store primitive data types (including strings) in JobDataMap to avoid data
 ### Use the Merged JobDataMap
 
 The JobDataMap that is found on the `JobExecutionContext` during Job execution serves as a convenience.
-It is a merge of the JobDataMap found on the JobDetail and the one found on the Trigger, with the value in the latter overriding any same-named values in the former.
+The data in the JobDataMap is a merger of the JobDetail and the Trigger, with the data in the Trigger overriding
```

Comment:
this is quite powerful concept, basically you can have one job and users could define in their settings that "I want daily digest". Would one model this with trigger per customer or one trigger or trigger per alphabet to split customer data processing to concurrent workloads that be run on different nodes, there are always tradeoffs.

Million triggers running at the same time is a lot  (trigger per user) - but always has per-trigger recovery, compared to handling all the hundreds (of thousands?) of emails to be sent and failing mid-process and retrying and sending multiples over and over again (did this once). State might need to be in application database (email processing queue with flags) + Quartz in this case.

Just a note.

Reply:
Yes! I want to have a page some place where we take your years of wisdom and try to distill it down into "things to think" about. As I'm working through the docs, I'm actively going back to my project and making improvements to the app and my underlying mental model.